assignments/callbacks.js

```diff
@@ -29,23 +29,78 @@ function getLength(arr, cb) {
   // getLength passes the length of the array into the callback.
 }
 
+function getLength(array, callback) {
+  callback(array.length);
+}
+getLength(items, (length) => {
+  alert(length);
+});
+// -------------------------------------------- //
+
 function last(arr, cb) {
   // last passes the last item of the array into the callback.
 }
 
+function last(arr, cb) {
+  return cb(arr[3]);
+}
+
+last(items, function(last) {
+  console.log(last)
+});
+// ------------------------------------------ //
+
+
 function sumNums(x, y, cb) {
   // sumNums adds two numbers (x, y) and passes the result to the callback.
 }
 
+function sumNums(x){
+  return function(y){
+    return x + y;
+  }
```

Comment:
This is a great example of closure, but the function needs to also accept a callback. Since you done one with multiply further down, I can show an alternate that follows the form you have here:

```js
function sumNumsWithCb(cb) {
    return function(x) {
        return function(y) {
            return cb(x + y);
        }
    }
}
```

As you can see, the surrounding functions provide closure for the values of `cb` and `x`, allowing the innermost nested function to return `cb(x + y);`